src/basic/calendarspec.c

```diff
@@ -876,6 +884,53 @@ static int parse_calendar_time(const char **p, CalendarSpec *c) {
         return r;
 }
 
+static int is_valid_timezone(char *tz, bool *result)
+{
+        pid_t pid;
+
+        if (tz == NULL || strlen(tz) == 0) {
+                *result = false;
+                return 0;
+        }
+
+        bool *shared = mmap(NULL, sizeof *shared, PROT_READ|PROT_WRITE, MAP_SHARED|MAP_ANONYMOUS, -1, 0);
+
+        if (shared == MAP_FAILED) {
+                return log_error_errno(errno, "mmap() failed: %m");
+        }
+
+        pid = fork();
+
+        if (pid == -1) {
+                munmap(shared, sizeof *shared);
+                return log_error_errno(errno, "Failed to fork: %m");
+        }
+
+        if (pid == 0) {
+                setenv("TZ", tz, 1);
+                tzset();
+
+                *shared = strlen(tzname[1]) > 0;
+
+                _exit(0);
+        }
+
+        if (waitpid(pid, NULL, 0) == -1) {
+                int waitpid_errno = errno;
+                munmap(shared, sizeof *shared);
+                return log_error_errno(waitpid_errno, "waitpid() failed: %m");;
+        }
+
+        bool tmp_result = *shared;
+        if (munmap(shared, sizeof *shared) != 0) {
+                return log_error_errno(errno, "munmap() failed: %m");;
+        }
+
+        *result = tmp_result;
```

Comment:
btw, using the mmap() thing appears unnecessary, for this the exit code of the child should be entirely enough for the information to transport